Fix unbind output

carolynvs opened the pull request:
This fixes the bug reported by @luksa in https://github.com/kubernetes-incubator/service-catalog/pull/2040#issuecomment-389874663.

* Add tests for svcat commands that support `--wait`.
* Stop using glog for any output we care about testing.
* Fix regression in `svcat unbind INSTANCE` where the binding names were not printed unless `--wait` was set.
* Add unit tests for unbind using the fake client. I plan on doing more of these tests for all of the commands in follow-on issues/PRs.

Here's the new output:

```console
$ svcat unbind --name mybinding
deleted mybinding

$ svcat unbind --name badbinding
remove binding default/badbinding failed

$ svcat unbind --name mybinding --wait
waiting for the binding to be deleted...
deleted mybinding

$ svcat unbind --name badbinding --wait
remove binding default/badbinding failed

$ svcat unbind myinstance
deleted binding2
deleted binding1

$ svcat unbind myinstance 
error:
  remove binding default/badbinding2 failed: sabotaged
deleted binding1
could not remove all bindings

$ svcat unbind myinstance --wait
error:
  remove binding default/badbinding2 failed: sabotaged
waiting for the bindings to be deleted...
deleted binding1
could not remove all bindings


```

luksa commented on cmd/svcat/binding/unbind_cmd.go:
```diff
@@ -134,6 +136,10 @@ func (c *unbindCmd) unbindInstance() error {
+	} else {
+		for _, binding := range bindings {
+			output.WriteDeletedResourceName(c.Output, binding.Name)
+		}
```
What if some bindings can't be deleted and some can? In that case, the method never gets to here because it immediately returns the error. The user sees only the error and there's no indication that some bindings were in fact deleted.

That was the reason I added this weird looking if statement: 
https://github.com/luksa/service-catalog/blob/97296b455101551cf52c52704439a1fcf2e3d525/cmd/svcat/binding/unbind_cmd.go#L107-L109

carolynvs replied:
Good catch, thank you! I'll get that updated.

carolynvs commented:
I am working on adding tests for this behavior now.

carolynvs commented:
I've added unit tests for unbind, and cleaned up the messages for consistency now that we are using the latest cobra release which formats errors differently. See the OP for what the messages look like now.

carolynvs commented:
@luksa Sorry this has grown so large. I finally have what I think is a much better way to test the cli figured out. Let me know what you think!

luksa commented:
/lgtm

luksa commented:
LGTM
